src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetService.cs

```diff
@@ -111,6 +112,13 @@ internal class DataSetService : IDataSetService
         };
     }
 
+    public async Task<Either<ActionResult, DataSetMetaViewModel>> GetMeta(Guid dataSetId, string? dataSetVersion = null)
+    {
+        return await GetVersionWithMeta(dataSetId, dataSetVersion)
+            .OnSuccessDo(_userService.CheckCanViewDataSetVersion)
```

Comment:
Good point.

I've refactored it to explicitly load the relationships (so that we don't have to load the version unnecessarily again), and have combined it with some `Includes` and with split queries.

I haven't yet tested it though because my migrations won't work :(

Reply:
Looks like using `.AsSplitQuery` is marginally worse (by like 0.1s out of an overall time of ~1.7s).

I've done some performance testing and have settled for the approach of assigning the relations myself with regular queries, and using `.AsNoTracking`. This was the fastest performance I could attain.

Reply:
However, the overall end-to-end response time now is around ~5-6 seconds for the 1st request, and ~1.8-2.5 seconds for subsequent requests...

I believe prior to these changes it was performing much closer to ~2 seconds for every request, even the first.

It may be that my machine is running slower today (which is certainly has been!) - but I might test this discrepancy on Monday to make sure we're not losing too much performance by doing this. Not sure what alternative we have though.

Reply:
It's much quicker now that we've improved the mapping efficiency. Back down to around 1.5-2s

Reply:
And yes, the mapping is what takes the majority of the time